File: FoodieApplication/searchService/src/main/java/com/example/searchService/model/Restaurant.java
package com.example.searchService.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@Document
public class Restaurant {

    @Id
    private String restaurantId;
    private String restaurantName;
    private String restaurantLocation;
    private List<Dish> dishList;
}

File: FoodieApplication/frontend/FoodieApplication/src/app/rest_management/domain/restaurant.ts
import { Dish } from "./dish";

export class Restaurant {
    
    restaurantId:string
    restaurantName:string
    restaurantLocation:string
    dishList: Array<Dish>

    constructor(){
        this.restaurantId='',
        this.restaurantName='';
        this.restaurantLocation='';
        this.dishList=[];
    }

}
